core/includes/common.inc

```diff
@@ -3630,13 +3630,8 @@ function backdrop_add_css($data = NULL, $options = NULL) {
   }
 
   // Construct the options, taking the defaults into consideration.
-  if (isset($options)) {
-    if (!is_array($options)) {
-      $options = array('type' => $options);
-    }
-  }
-  else {
-    $options = array();
+  if (isset($options) && !is_array($options)) {
```

Comment:
This line could be probably changed to `if (isset($options) && is_string($options)) {`. The `'options'` key is expected to be a string, not an object or a resource, for example.